### ISSUE TITLE
[$500] Desktop - Magic code link opens the app, but it glitches and returns to the login screen

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#32783]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.10-1
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
Prerequisites: User must be logged into web app as User A.
1. Launch the Desktop app
2. Enter an email for an existing or new account
3. Go to the email inbox and locate the magic code
4. Copy the Magic link and modify it to staging
5. On the web, open a new tab and navigate to the staging magic link
6. Confirm that the user is prompted to open the desktop app
7. Click on the "Open app" option

## Expected Result:
When redirected to the desktop app the user must be logged into the new or existing account and no visual problems should be observed

## Actual Result:
When the desktop is launched by the magic code link, it glitches, briefly shows a skeleton and then returns to the first login screen

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/76caabce-f4ce-4775-bbd9-3aee3b237459

[logs.txt](https://github.com/Expensify/App/files/13629766/logs.txt)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0190046c73ec67f0a3</li>
        <li>Upwork Job ID: 1734155185383477248</li>
        <li>Last Price Increase: 2024-01-04</li>
    </ul>
</details>

**Comment by Julesssss** [comment]
I'm not convinced that this does work on Prod. First I see a 'Oops this page cannot be found' message, then when I refresh I am taken back to the authentication page.

**Comment by Julesssss** [comment]
Lets see if we can locate the regression, but this shouldn't block the release because: 
- You cannot currently sign in on production using the link
- It only occurs on Desktop

**Comment by Julesssss** [comment]
Closing in favour of [this ](https://github.com/Expensify/App/issues/32795)similar, but more descriptive issue

**Comment by kavimuru** [comment]
Issue is reproduced today as well. So reopening the issue.

**Comment by muttmuure** [comment]
Looking at higher priority items first

**Comment by muttmuure** [comment]
I'm not able to reproduce this because when I put the link into Chrome it doesn't give me the option to open desktop at all

**Comment by muttmuure** [comment]
Can't reproduce this. What I am seeing is: 

1. Generate magic code from desktop app
2. Copy paste staging.new.expensify.com link into browser
3. The page opens to the magic sign in code. (There is no option to open Desktop).

You can opt to sign in there with the magic link, and then using the link to authorize a desktop sign in is expired. However, that seems like a separate problem
